### DISCUSSION TITLE
Persist Session

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [zauberzeug/nicegui#498] [Q&A] [answered]
I implemented sessions It works great. 

Now I want to persist the session, so I quickly made a subclass of `SessionMiddleware ` to see if I could override some method
to intercept the session and store it let's say in a .json file (to later save it in a MongoDB database).

The only method I see it's `__call__ ` and I don't want to mess with all that code, so I'm asking the experts what's the best practice in nicegui to persist the session?

Thank you!

**Comment by rodja** [comment, marked as answer]
By default the SessionMiddleware stores the data in the cookie. So you normally do not need to persist anything. For security reasons you may want to only store a session-id in the cookie and then have a lookup in a local database for the session state. We mostly use Postgresql with SqlAlchemy to do this. But you could also use any other form of persistence.

Note that you do not need to derive from SessionMiddleware to do so. Simply do your database query with the session id wherever you need the session data.

**Reply by angelhdzmultimedia** [reply]
Yes, that's how I do it in `Node w/ Express` and `express-session`. I store a token, and use the token to look up the user in the DB.
Then I use `connect-mongodb-session` to persist the session as well in the database in another collection.

Now I understand how to work with nicegui and sessions, thank you!

This is all promising!
